examples/official-storybook/stories/addon-docs/props.stories.mdx

```diff
@@ -87,12 +97,28 @@ export const FooBar = ({ foo, bar, baz } = {}) => (
 <Preview>
   <Story
     name="Controls"
+    args={{
+      background: '#ff0',
+      label: 'apple',
+      count: 5,
+    }}
     argTypes={{
-      foo: { control: { type: 'range' } },
-      bar: { control: { type: 'options', options: ['a', 'b', 'c'] } },
+      count: { control: { type: 'range', min: 0, max: 10 } },
+      label: {
+        control: { type: 'options', options: ['apple', 'banana', 'cherry'] },
```

Comment:
what are the other properties here?

why:
`argTypes[name].control`

vs
`argTypes[name]`

?

Reply:
- `argTypes[name]` = the ArgType
- `argTypes[name].x` = addon-specific annotations

Reply:
But there's nothing else in ArgTypes[name] ?

> what are the other properties here?

```
<Preview>
  <Story
    name="ArgTypes"
    args={{
      foo: false,
      bar: '',
      baz: ['a', 'b'],
    }}
    argTypes={{
      foo: { name: 'foo', type: { name: 'boolean' }, description: 'foo description' },
      bar: { name: 'bar', type: { name: 'string' }, description: 'bar description' },
      baz: {
        name: 'baz',
        type: { name: 'array', value: { name: 'string' } },
        description: 'baz description',
      },
    }}
  >
    {(args) => <FooBar {...args} />}
  </Story>
</Preview>
```

vs

```
<Preview>
  <Story
    name="Controls"
    args={{
      background: '#ff0',
      label: 'apple',
      count: 5,
    }}
    argTypes={{
      count: { control: { type: 'range', min: 0, max: 10 } },
      label: {
        control: { type: 'options', options: ['apple', 'banana', 'cherry'] },
      },
      background: { control: { type: 'color' } },
    }}
  >
    {({ background, label, count }) => {
      return (
        <ButtonGroup background={background}>
          {range(count).map((i) => (
            <DocgenButton label={`${label} ${i}`} />
          ))}
        </ButtonGroup>
      );
    }}
  </Story>
</Preview>
```

Seems like there so much overlap here, to me.

I find it odd we have 2 `type` properties

I'd actually expect something like this:
```
baz: {
  name: 'baz',
  type: 'oneOf',
  value: 'apple',
  possibleValues: ['apple', 'banana', 'cherry'],
  control: 'select',
  description: 'baz description',
},
```

Reply:
Most of that stuff gets filled in automatically based on propTypes / typescript types